src/gui/state/game_menu.ts

```diff
@@ -0,0 +1,134 @@
+import type Graphics from "../../Graphics.js"
+import MenuState from "../../enums/MenuState.js"
+import { game, getTexture, menuState, setCurrentWorldName, setMenuState } from "../../main.js"
+import TextRenderer from "../../util/TextRenderer.js"
+import { Button } from "../Button.js"
+import * as ingame_state from "../../gui/state/ingame.js"
+import YSON from "https://j0code.github.io/browserjs-yson/main.mjs"
+import World from "../../world/World.js"
+import Texture from "../../texture/Texture.js"
+
+let widgetsTex
+let logoTex: Texture
+const singleplayerButton = new Button(0, 200, 800, 80, "Singleplayer")
+const optionsButton = new Button(0, 300, 800, 80, "Options")
+const quitButton = new Button(0, 400, 800, 80, "Quit")
+let worldButtons: Button[] = []
+const createWorldButton = new Button(0, 100, 800, 80, "Create World")
+
+singleplayerButton.on("click", () => {
+	worldButtons = []
+	const worlds = JSON.parse(localStorage.getItem("worlds") || "[]");
+	for (let i = 0; i < worlds.length; i++) {
+		const world = worlds[i]
+		const button = new Button(0, 200 + i * 100, 800, 80, world.name)
+		button.on("click", () => {
+			console.log(JSON.stringify(world));
+			setCurrentWorldName(world.name)
+			const worldObj = World.load(world.stringBlocks, world.blockData, world.dims, world.entities)
+			if(!worldObj) {
+				alert("Failed to load world!")
+				return
+			}
+			ingame_state.init()
+			ingame_state.setWorld(worldObj)
+			setMenuState(MenuState.INGAME)
+			worldObj.spawn(ingame_state.player)
+		})
+		worldButtons.push(button)
+	}
+	setMenuState(MenuState.WORLDSELECTION);
+})
+
+optionsButton.on("click", () => {
+	console.log("button 2 clicked")
+})
+
+quitButton.on("click", () => {
+	location.href = "about:blank";
```

Comment:
```suggestion
	try {
		window.close()
	} catch (e) {
		location.href = "about:blank"
	}
```
I know this doesn't work normally, but in PWA it will.
Your code as fallback.